x/ibc/light-clients/solomachine/types/solomachine.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"github.com/tendermint/tendermint/crypto"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+)
+
+var _, _, _, _ codectypes.UnpackInterfacesMessage = &ClientState{}, &ConsensusState{}, &Header{}, &HeaderData{}
```

Comment:
ditto, not sure if this will do a correct type assertion.

Also, let's add a comment.